File: app/src/main/java/com/shindra/TramViewHolder.java
package com.shindra;

import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.shindra.ctslibrary.bo.Line;


public class TramViewHolder extends RecyclerView.ViewHolder  {

    ImageView mImageView;
    Button mTimeButton;

    public TramViewHolder(@NonNull View pItemView) {

        super(pItemView);
        mImageView = itemView.findViewById(R.id.TramText);
        this.mTimeButton = itemView.findViewById(R.id.TimeButton);
    }

    public void onBind(Line pLine, RecyclerItemClick pCallback){

        mTimeButton.setOnClickListener(v -> pCallback.onTimeClick(pLine));
    }
}

File: app/src/main/java/com/shindra/Tram_ViewHolder.kt
package com.shindra

import android.view.View
import android.widget.Button
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import com.shindra.ctslibrary.bo.Line

class Tram_ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    val tramIcon: ImageView = itemView.findViewById(R.id.image_tram)
    val scheduleButton: Button = itemView.findViewById(R.id.schedule_btn)

    fun onBind(line: Line, callback: OnLineClickListener) {
        scheduleButton.setOnClickListener{
            callback.onItemClick(line)
        }
    }

    interface OnLineClickListener{
        fun onItemClick(line: Line)
    }
}